torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -21,22 +21,23 @@ const inline char* getNcclErrorDetailStr(ncclResult_t error, c10::optional<std::
   if (processGroupFailureReason != c10::nullopt) {
     return (*processGroupFailureReason).c_str();
   }
+  std::string err = std::string(ncclGetLastError(NULL));
```

Comment:
Do we need to check the NCCL version before calling this API? https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/api/comms.html#ncclgetversion

Reply:
Given that our nccl version both in OSS and internal already have this API, I think we should be fine since this code will ship together with the newer nccl version?

Reply:
It does look like jobs in https://www.internalfb.com/diff/D39161199 are failing due to this new function call. I noticed that we can use a #if directive to describe using this feature. I will add example, in a separate comment.